### PR TITLE
Update default layer colors

### LINKED ISSUE #406: Improve default layer colors
##### SUMMARY

The current default layer colors are following (small demo project created [here](https://github.com/LibrePCB-Projects/layer-colors)):

grafik

But those colors aren't ideal. Concrete issues:
- Many layers have exactly the same color.
  - Placement, names and values (which are typically used for silkscreen) cannot be distinguished from documentation layers. Thus it's not visible what's printed on silkscreen and what not.
- Bottom copper has very bad contrast to the background and thus is not very readable.

##### SOLUTION

Find good layer colors and adjust `libs/librepcb/common/graphics/graphicslayer.cpp` accordingly. The small test project mentioned above can be used to test the layer colors. 

Note: It's probably easier to play around with the layer colors directly in the project's file `boards/default/settings.user.lp` instead of adjusting the source code and compile after every change. The layer colors in the source code are only used as default anyway, if the board settings file exists, those layer colors will be used instead.

### PR DESCRIPTION
**Pull request by dbrgn**
Still untested, I hope it compiles :slightly_smiling_face: 

Fixes #406.

### CODE REVIEW

File: libs/librepcb/common/graphics/graphicslayer.cpp [hunk 1]
```diff
@@ -214,15 +214,15 @@ void GraphicsLayer::getDefaultValues(const QString& name, QString& nameTr,
     // clang-format off
     // schematic
     h.insert(sSchematicReferences,      {tr("References"),            QColor(0, 0, 0, 50),        QColor(0, 0, 0, 80),        true});
-    h.insert(sSchematicSheetFrames,     {tr("Sheet Frames"),          Qt::black,                  Qt::darkGray,               true});
+    h.insert(sSchematicSheetFrames,     {tr("Sheet Frames"),          QColor("#96E0E0E0"),        QColor("#FFFFFFFF"),        true});
```
Comment (ubruhin) [inline libs/librepcb/common/graphics/graphicslayer.cpp]: Are you sure you want to modify the **schematic** layer? :wink:

File: libs/librepcb/common/graphics/graphicslayer.cpp [hunk 2]
```diff
@@ -214,15 +214,15 @@ void GraphicsLayer::getDefaultValues(const QString& name, QString& nameTr,
...
     h.insert(sSchematicNetLines,        {tr("Netlines"),              Qt::darkGreen,              Qt::green,                  true});
     h.insert(sSchematicNetLabels,       {tr("Netlabels"),             Qt::darkGreen,              Qt::green,                  true});
     h.insert(sSchematicNetLabelAnchors, {tr("Netlabel Anchors"),      Qt::darkGray,               Qt::gray,                   true});
     h.insert(sSchematicDocumentation,   {tr("Documentation"),         Qt::darkGray,               Qt::gray,                   true});
     h.insert(sSchematicComments,        {tr("Comments"),              Qt::darkBlue,               Qt::blue,                   true});
     h.insert(sSchematicGuide,           {tr("Guide"),                 Qt::darkYellow,             Qt::yellow,                 true});
     // symbol
-    h.insert(sSymbolOutlines,           {tr("Outlines"),              Qt::darkRed,                Qt::red,                    true});
+    h.insert(sSymbolOutlines,           {tr("Outlines"),              QColor("#DDFFFFFF"),        QColor("#FFFFFFFF"),        true});
```
Comment (ubruhin) [inline libs/librepcb/common/graphics/graphicslayer.cpp]: Same here.

File: libs/librepcb/common/graphics/graphicslayer.cpp (hunk 1 above)
Comment (dbrgn) [reply libs/librepcb/common/graphics/graphicslayer.cpp]: Whoopsie.

**Comment by dbrgn** [comment]
@ubruhin ping!

File: libs/librepcb/common/graphics/graphicslayer.cpp [hunk 3]
```diff
@@ -232,54 +232,82 @@ void GraphicsLayer::getDefaultValues(const QString& name, QString& nameTr,
...
-    h.insert(sBoardOutlines,            {tr("Board Outlines"),        QColor(255, 255, 255, 180), QColor(255, 255, 255, 220), true});
-    h.insert(sBoardMillingPth,          {tr("Milling (PTH)"),         QColor(0, 255, 255, 150),   QColor(0, 255, 255, 220),   true});
-    h.insert(sBoardDrillsNpth,          {tr("Drills (NPTH)"),         QColor(255, 255, 255, 200), QColor(255, 255, 255, 255), true});
-    h.insert(sBoardPadsTht,             {tr("Pads"),                  QColor(0, 255, 0, 150),     QColor(0, 255, 0, 220),     true});
-    h.insert(sBoardViasTht,             {tr("Vias"),                  QColor(0, 255, 0, 150),     QColor(0, 255, 0, 220),     true});
+    h.insert(sBoardSheetFrames,         {tr("Sheet Frames"),          QColor("#96E0E0E0"),        QColor("#FFFFFFFF"),        true});
+    h.insert(sBoardOutlines,            {tr("Board Outlines"),        QColor("#C8FFFFFF"),        QColor("#FFFFFFFF"),        true});
+    h.insert(sBoardMillingPth,          {tr("Milling (PTH)"),         QColor("#C8FFFFFF"),        QColor("#FFFFFFFF"),        true});
```
Comment (ubruhin) [inline libs/librepcb/common/graphics/graphicslayer.cpp]: I would suggest 00DDFF and 00FFFF to make it distinguishable from board outlines:

```suggestion
    h.insert(sBoardMillingPth,          {tr("Milling (PTH)"),         QColor("#C800DDFF"),        QColor("#FF00FFFF"),        true});
```

File: libs/librepcb/common/graphics/graphicslayer.cpp [hunk 4]
```diff
@@ -232,54 +232,82 @@ void GraphicsLayer::getDefaultValues(const QString& name, QString& nameTr,
...
-    h.insert(sBoardMeasures,            {tr("Measures"),              Qt::gray,                   Qt::lightGray,              true});
-    h.insert(sBoardAlignment,           {tr("Alignment"),             Qt::darkCyan,               Qt::cyan,                   true});
-    h.insert(sBoardDocumentation,       {tr("Documentation"),         Qt::lightGray,              Qt::white,                  true});
-    h.insert(sBoardComments,            {tr("Comments"),              QColor(255, 165, 0, 180),   QColor(255, 165, 0, 255),   true});
-    h.insert(sBoardGuide,               {tr("Guide"),                 Qt::darkYellow,             Qt::yellow,                 true});
+    h.insert(sBoardMeasures,            {tr("Measures"),              QColor("#FF808000"),        QColor("#FFA3B200"),        true});
+    h.insert(sBoardAlignment,           {tr("Alignment"),             QColor("#B4E59500"),        QColor("#DCFFBF00"),        true});
+    h.insert(sBoardDocumentation,       {tr("Documentation"),         QColor("#B4E59500"),        QColor("#DCFFBF00"),        true});
```
Comment (ubruhin) [inline libs/librepcb/common/graphics/graphicslayer.cpp]: IMHO this layer looks better when using the same color as for top and bottom documentation:

```suggestion
    h.insert(sBoardDocumentation,       {tr("Documentation"),         QColor("#96E0E0E0"),        QColor("#DCE0E0E0"),        true});
```

**Comment by ubruhin** [comment]
@dbrgn What do you think about the two suggestions? Go or no-go? :)

File: libs/librepcb/common/graphics/graphicslayer.cpp (hunk 4 above)
Comment (dbrgn) [reply libs/librepcb/common/graphics/graphicslayer.cpp]: Do it! (I can't commit the suggestion for some reason...)

File: libs/librepcb/common/graphics/graphicslayer.cpp (hunk 3 above)
Comment (dbrgn) [reply libs/librepcb/common/graphics/graphicslayer.cpp]: Looks good!

**Pull request merged**